snuba/request/request_query_settings.py

```diff
@@ -0,0 +1,26 @@
+from typing import Sequence
+
+from snuba.state.rate_limit import get_global_rate_limit_params, RateLimitParameters
+
+
+class RequestQuerySettings:
```

Comment:
Why the renaming from QuerySettings and RequedtQuerySettings.
Also could you please provide a docstring for this. It would be useful to give a good description of the role and the scope of this class.

Reply:
`RequestSettings`, to me, are settings on how a request should be run. `RequestQuerySettings` are settings on the query in the request. With rate limits, it's no longer just the settings on the request, but settings on the query in the request. Does it seem odd?

Yea, I'll add docstrings :+1:

Reply:
_With rate limits, it's no longer just the settings on the request, but settings on the query in the request_

Why so? It seems to me that as of now rate limiting applies to the request, not to the query.

Also please beef up the docstring explaining what the difference is between what kind of information goes here and what should go in the Query class. I think trying to define this boundary would be a good exercise to define the concept in the architecture

Reply:
Hmmm, I suppose I can see it from both sides. We are rate limiting on the queries in CH but as a result, also rate limiting the request.

Either way, I don't have a strong opinion so, I'll revert the rename.